javascript/safari-driver/inject/commands/commands.js

```diff
@@ -97,6 +97,20 @@ safaridriver.inject.commands.getTitle = function() {
 };
 
 
+/**  @param {function(string)} fn The function to call when the data is ready. */
+safaridriver.inject.commands.getCanvasUrl = function(callback) {
```

Comment:
Use a promise, not a callback.

    getCanvasUrl = function() {
      return new webdriver.promise.Promise(function(fulfill) {
          html2canvas(document.body, {
            onrendered: function (canvas) {
              canvasDataURL = canvas.toDataURL();
              fulfill(canvasDataURL);
            }
          });
      });
    };